python/rgbmatrix/core.pyx

```diff
@@ -83,18 +83,100 @@ cdef class FrameCanvas(Canvas):
+        def __set__(self, uint32_t value): self.__options.pwm_lsb_nanoseconds = value
+
+    property brightness:
+        def __get__(self): return self.__options.brightness
+        def __set__(self, uint8_t value): self.__options.brightness = value
+
+    property disable_hardware_pulsing:
+        def __get__(self): return self.__options.disable_hardware_pulsing
+        def __set__(self, value): self.__options.disable_hardware_pulsing = value
+
+    property show_refresh_rate:
+        def __get__(self): return self.__options.show_refresh_rate
+        def __set__(self, value): self.__options.show_refresh_rate = value
+
+    property swap_green_blue:
+        def __get__(self): return self.__options.swap_green_blue
+        def __set__(self, value): self.__options.swap_green_blue = value
+
+    property inverse_colors:
+        def __get__(self): return self.__options.inverse_colors
+        def __set__(self, value): self.__options.inverse_colors = value
+
+    # RuntimeOptions properties
+
+    property gpio_slowdown:
+        def __get__(self): return self.__runtime_options.gpio_slowdown
+        def __set__(self, uint8_t value): self.__runtime_options.gpio_slowdown = value
+
+    property daemon:
+        def __get__(self): return self.__runtime_options.daemon
+        def __set__(self, uint8_t value): self.__runtime_options.daemon = value
+
+    property drop_privileges:
+        def __get__(self): return self.__runtime_options.drop_privileges
+        def __set__(self, uint8_t value): self.__runtime_options.drop_privileges = value
+
 
 cdef class RGBMatrix(Canvas):
-    def __cinit__(self, int rows, int chains = 1, int parallel = 1):
-        # TODO(Saij): this should initialize an RGBMatrix::Options and
-        # RuntimeOptions, then call CreateMatrixFromOptions() instead of the
-        # cppinc.RGBMatrix() constructor directly. No __gpio needed anymore.
-        # The options allow to set more things, so they should probably be
-        # available as named parameters in Python ?
+    def __cinit__(self, int rows = 0, int chains = 0, int parallel = 0,
+        RGBMatrixOptions options = None):
+
+        # If RGBMatrixOptions not provided, create defaults and set any optional
+        # parameters supplied
+        if options == None:
+            options = RGBMatrixOptions()
+            if rows > 0:
+                options.rows = rows
+            if chains > 0:
+                options.chain_length = chains
+            if parallel > 0:
+                options.parallel = parallel
+        # If RGBMatrixOptions provided with also any of rows/chains/parallel
+        # use either RGBMatrixOptions OR rows/chains/parallel
+        elif options != None and (rows > 0 or chains > 0 or parallel > 0):
+            raise Exception("Providing RGBMatrixOptions with rows/chains/parallel parameters not supported.")
+
+
+        # For backward compatibility?
         self.__gpio = new cppinc.GPIO()
```

Comment:
I left it for backward compatibility, it case someone uses this to read the GPIO value.